### PR TITLE
[5.5] Fix: verify csrf token allows to except full urls

### PR DESCRIPTION
**Pull request by BenCavens**
Currently, you can leave out certain URLs from CSRF token verification by adding them to the `except` array. This is provided with an out-of-the-box laravel application. 

However, this only works if you add the paths and not the full URLs. This PR makes sure it is also possible to provide full URLs to this except array.

### CODE REVIEW

File: src/Illuminate/Foundation/Http/Middleware/VerifyCsrfToken.php [hunk 1]
```diff
@@ -104,7 +104,7 @@ protected function inExceptArray($request)
                 $except = trim($except, '/');
             }
 
-            if ($request->is($except)) {
+            if ($request->url() == $except || $request->is($except)) {
```
Comment (laurencei) [inline src/Illuminate/Foundation/Http/Middleware/VerifyCsrfToken.php]: should this be `===`?

**Comment by vlakoff** [comment]
Would it need wildcards support as well? For consistency with the previsouly existing `$request->is()`.

**Comment by sisve** [comment]
I believe this PR is missing tests. Could you add some?

**Comment by tillkruss** [comment]
@BenCavens: Please add a test.

**Comment by BenCavens** [comment]
I'll try to add some tests. 
It's not so straightforward because currently the verifytoken middleware isn't tested at all.

**Comment by vlakoff** [comment]
Can you please tell if wildcard support should be added? (I'd bend supposing it should be)

Also, may I ask for a use case, where you have to use full urls instead of paths?

**Comment by BenCavens** [comment]
@vlakoff @tillkruss I've created a new PR which adds the wildcard support as well as the necessary tests: https://github.com/laravel/framework/pull/22661 

I'm looking forward to your feedback. thx

**Comment by tillkruss** [comment]
@BenCavens: Thanks, great work!

**Pull request merged**